File: app/src/main/java/com/permissionnanny/missioncontrol/AppControlActivity.java
package com.permissionnanny.missioncontrol;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import com.permissionnanny.BaseActivity;
import javax.inject.Inject;

public class AppControlActivity extends BaseActivity {

    @Inject AppControlBinder mBinder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getComponent().inject(this);
        mBinder.onCreate(savedInstanceState);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return mBinder.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return mBinder.onOptionsItemSelected(item);
    }

    @Override
    protected void onResume() {
        super.onResume();
        mBinder.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mBinder.onPause();
    }
}

File: app/src/main/kotlin/com/permissionnanny/missioncontrol/AppControlActivity.kt
package com.permissionnanny.missioncontrol

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.permissionnanny.BaseActivity
import javax.inject.Inject

open class AppControlActivity : BaseActivity() {

    @Inject internal lateinit var binder: AppControlBinder

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getComponent().inject(this)
        binder.onCreate(savedInstanceState)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        return binder.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return binder.onOptionsItemSelected(item)
    }

    override fun onResume() {
        super.onResume()
        binder.onResume()
    }

    override fun onPause() {
        super.onPause()
        binder.onPause()
    }
}
